roles/openshift_health_checker/openshift_checks/sdn.py

```diff
@@ -0,0 +1,698 @@
+"""
+Check that the SDN is routing traffic properly.
+"""
+
+import base64
+import os
+import re
+
+import yaml
+from ansible.module_utils import six
+
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException
+from openshift_checks.mixins import NotContainerizedMixin
+
+
+class SDNCheck(NotContainerizedMixin, OpenShiftCheck):
+    """A check to run relevant diagnostics on the SDN."""
+
+    name = 'sdn'
+    tags = ['health']
+
+    def is_active(self):
+        """Skip hosts that are not masters or nodes."""
+        group_names = self.get_var('group_names', default=[])
+        master_or_node = 'oo_masters_to_config' in group_names or \
+                         'oo_nodes_to_config' in group_names
+        return super(SDNCheck, self).is_active() and master_or_node
+
+    def run(self):
+        if self.want_full_results:
+            # Gather diagnostic information and perform diagnostics on a master
+            # or node host.
+            try:
+                # Boot log can take too long to load on some systems.
+                # self.save_command_output('journal-full.log', ['journalctl',
+                #                                               '--boot'])
+                self.save_command_output('nmcli-dev',
+                                         ['nmcli', '--nocheck', '-f', 'all',
+                                          'dev', 'show'])
+                self.save_command_output('nmcli-con',
+                                         ['nmcli', '--nocheck', '-f', 'all',
+                                          'con', 'show'])
+                self.save_command_output(
+                    'ifcfg',
+                    'head -1000 /etc/sysconfig/network-scripts/ifcfg-*')
+                self.save_command_output('addresses', ['ip', 'addr', 'show'])
+                self.save_command_output('routes', ['ip', 'route', 'show'])
+                self.save_command_output('arp',
+                                         ['ip', '-s', 'neighbor', 'show'])
+                # XXX: Shouldn't we use iptables -nL in order to preserve more
+                # information?
+                self.save_command_output('iptables', ['iptables-save'])
+                self.register_file('hosts', None, '/etc/hosts')
+                self.register_file('resolv.conf', None, '/etc/resolv.conf')
+                self.save_command_output('modules', ['lsmod'])
+                self.save_command_output('sysctl', ['sysctl', '-a'])
+                # XXX: Would self.get_var('openshift', 'docker') suffice?
```

Comment:
I don't know the details of the ansible stuff, but yeah, some of this info may have already been gathered by other modules (and if so, and it gets saved somewhere else already, we don't even need to save it here).